### ISSUE TITLE
Why people are turning away from this package?

### ISSUE THREAD

**Issue by Compro-Prasad** [auto-complete/auto-complete#503]
I used to use it a year ago but seeing people switching to company-mode makes me feel uncomfortable. What is it that auto-complete isn't providing because of which people are switching to company-mode?

**Comment by mpfeifer** [comment]
Maybe the Original owner of this Repository hast turned away for some reasons.  Btw its always possible to fork the project under more active ownership :-)

**Comment by articuluxe** [comment]
It does seem that company is more actively maintained. 

I stick with auto-complete because its terminal support is IMO better than company; and tooltips seem to work better than company (though it’s been a while since I did the comparison).

**Comment by grettke** [comment]
It still use it and it seems fine.

For me it is worth picking up if it quits working.

**Comment by ylluminarious** [comment]
Seems to mainly boil down to hype and marketing. Lots of people online are very vocal about company being "better." In some regards, this is true. In other regards, I think the packages are basically on par with each other.

One important thing to note is that [company is in GNU ELPA](https://elpa.gnu.org/packages/company.html) which helps visibility a lot and gives some assurance that it probably won't ever get too awfully stale.

I would like to address some technical points, though. Among other things, people often claim that company has:

1. cleaner code
2. less bugs
3. better API

My response to these claims, in order:

1. It probably has cleaner code, since large swathes of `popup.el` (the library that auto-complete relies on) are undocumented and hard to read.
2. Fewer bugs? Meh, maybe. I haven't seen company being much less buggy than auto-complete.
3. Better API? Perhaps. There is certainly more community activity going on for company which can certainly help that sort of thing.

However, one important detail regarding community involvement that should not be neglected is that company is an official GNU package. This means that one has to sign [copyright assignment papers](https://www.gnu.org/prep/maintain/html_node/Copyright-Papers.html) to the FSF before contributing to company. Just an important detail to keep in mind.

Personally, I use auto-complete because I found it first (due to the much more intuitive name). I'm already entrenched in it since I have other packages which rely on auto-complete and it works well enough for daily usage. There are some annoying things and the lack of community support has put folks like me on a dead-end road. We'll all probably have to switch to company someday due to the auto-complete developers' abandonment of the project.

There is one actual issue, though, that stands out in both of the projects to me. So far, only a few others seem to care about this, but I'm surprised that it's not more urgent to the developers of either project. Zooming in on a buffer breaks the completion popup, in both company and auto-complete. I've spoken about this [here](https://github.com/auto-complete/popup-el/issues/6#issuecomment-303977423) and [here.](https://github.com/company-mode/company-mode/issues/299#issuecomment-372920120) The only thing that exists to address this issue is ["company-postframe"](https://github.com/tumashu/company-posframe) (gotta love these unintuitive names). But this is quite experimental and probably brittle.

Aside from this singular issue, both of these packages seem mostly the same from an end-user perspective. I really wish auto-complete would get forked and revitalized so that people like me don't have to declare auto-completion bankruptcy and shuffle around large parts of our configs someday. But maintaining a fork of this project would take a lot of time and effort, which I do not have.

If there is someone out there who could take up the torch, you'd certainly have a lot of people who'd appreciate it.

**Comment by jonnycyr** [comment]
I'm a company-mode user, I've only tried auto-complete a tiny bit - so I can't really give any functionality comparisons. To add onto what @ylluminarious said I think an important thing to note is that auto-complete has documentation while company-mode does not.

This may not be a big deal for some people - since you can always look through code and read the docstrings, but I think documentation is incredibly important because it can contain extra tidbits that docstrings and comments may not have. Company-mode gives one the ability to add great amounts of customization - and it'd be great if there was accompany documentation show some examples of this. Instead, you're forced to either look through the company code or look at other peoples code. If you're someone just getting started with emacs that's a daunting task to take on. 

Heck, to get completion working in emacs you basically have two choices:
- use a package that hasn't been updated in a while 
- or use a package that has no documentation so you have to look through code 

If auto-complete development was still active I'd definitely be an auto-complete user than a company-mode user.

**Comment by ylluminarious** [comment]
@jonnycyr I completely agree. Auto-complete's documentation is something that I neglected to mention. Actually, auto-complete's documentation has helped me quite a lot in the past. You are right; company-mode's lack of documentation is a serious deficiency. It's important for beginners to have things like info manuals or just anything more than docstrings. Even for long-time Emacs users, manuals are still quite helpful. English is just easier for our brains to parse than Lisp, if for no other reason than it is what we speak.

**Comment by coretemp** [comment]
Considering that it doesn't build, it seems I have little choice than to not use it. So, that might be a reason for people to find something else (or to fix it). 

```
In toplevel form:
fuzzy.el:36:1:Warning: defcustom for `fuzzy-accept-error-rate' fails to
    specify type
fuzzy.el:36:1:Warning: defcustom for `fuzzy-accept-error-rate' fails to
    specify type

In fuzzy-regexp-compile:
fuzzy.el:51:30:Warning: `labels' is an obsolete macro (as of 24.3); use
    `cl-labels' instead.

In fuzzy-isearch:
fuzzy.el:217:10:Warning: `isearch-word' is an obsolete variable (as of 25.1);
    use `isearch-regexp-function' instead.
make: *** [Makefile:5: byte-compile] Error 1
```

**Comment by ylluminarious** [comment]
@coretemp Hmm, that's strange. My installation of auto-complete was compiled with Emacs 26.1 a few months ago and I had no such errors which prevented compilation.

**Comment by coretemp** [comment]
@ylluminarious Yes, I wrote the comment a bit too early. I got it to work by changing package source.

**Comment by ylluminarious** [comment]
@coretemp Ok, gotcha.

**Comment by zw963** [comment]
In fact, i use emacs-git (version 27), auto-complete still work well for me.

```
Personally, I use auto-complete because I found it first (due to the much more intuitive name). I'm already entrenched in it since I have other packages which rely on auto-complete and it works well enough for daily usage. 
```

In fact,  this is the same reason for me still keep use auto-complete, but the fact is, some new created package was not support auto-complete anymore, it just support company-mode, we have to find out a 3-party package to working with those new package.

**Comment by ylluminarious** [comment]
@zw963 Sorry for your predicament. Some months ago, I faced the same problem, so I just bit the bullet and moved to company-mode. I'm not sure if auto-complete has much of a future anymore, unless someone takes up the torch and continues maintaining the project. It's just a matter of time until it stops working altogether. It's hard to keep using a library which is being deprecated and dropped by most other packages.

**Comment by tarsius** [comment]
Please note that this repository is now looking for a new maintainer. See #509.

**Comment by ylluminarious** [comment]
@tarsius Maybe announce this on /r/emacs or emacs-devel to get some visibility? I'm not volunteering for the position myself at this time.

**Comment by dickmao** [comment]
> Why people are turning away from this package?

There is little to recommend auto-complete over company.  It's only slightly faster in certain contexts but its shortcomings in capability and extensibility (hardcoded keywords, etc.) is a direct reflection of a deficit of man-hours spent, not the quality of the software.

**Comment by ArneBab** [comment]
I prefer auto-complete, because it feels snappier and the overlays in the terminal are great. The one problem I had with it when completing in mu4e turned out to also appear in company-mode — but I had already switched to company mode when I realized that.

Seeing that auto-complete isn’t completely unmaintained anymore, I might switch back to auto-complete soon.
